ui/app/main.js

```diff
@@ -1,82 +1,131 @@
+            <div class="row">
+              <div class="col-sm-6">
+                <div class="panel panel-default panel-body">
+                  <p>
+                    <a href="/auth/login" class="btn btn-primary btn-block btn-lg"><i class="fa fa-external-link"></i> Authorize using redirect</a>
+                  </p>
+                  <p>
+                    The user authorizes the application after being redirected to the Blackbaud Authorization website.
+                  </p>
+                </div>
+              </div>
+              <div class="col-sm-6">
+                <div class="panel panel-default panel-body">
+                  <p>
+                    <button id="popup-login" class="btn btn-primary btn-block btn-lg" target="login-iframe"><i class="fa fa-window-restore"></i> Authorize using popup</button>
+                  </p>
+                  <p>
+                    The user authorizes the application using a browser popup window.
+                  </p>
+                </div>
+              </div>
+            </div>
+          </div>
+        `;
+        app.innerHTML = loginOptions;
 
-                popup = $window.open('auth/login?redirect=/%23/auth-success', 'login', 'height=450,width=600');
+        // Add event listener to popup login button
+        document.getElementById("popup-login").addEventListener("click", function () {
+            const popup = window.open('auth/login?redirect=/%23/auth-success', 'login', 'height=450,width=600');
+            if (window.focus) {
+                popup.focus();
+            }
 
-                if ($window.focus) {
-                    popup.focus();
-                }
-            };
+            let intervalId = setInterval(function () {
+                fetch('/auth/authenticated')
+                    .then(response => response.json())
+                    .then(data => {
+                        if (data.authenticated) {
+                            clearInterval(intervalId);
+                            popup.close();
+                            // User is authenticated, fetch constituent data
+                            fetch('/api/constituents/280')
+                                .then(response => response.json())
+                                .then(data => {
+                                    const constituent = data;
+                                    renderConstituentData(constituent);
+                                });
+                        }
+                    });
+            }, 200);
         });
-})(window.angular);
+    }
+
+    // Render constituent data
+    function renderConstituentData(constituent) {
+        const constituentData = `
+        <h1>SKY API Authorization Code Flow Tutorial</h1>
+          <div class="well">
+            <h3 class="well-title">Constituent: ${constituent.name}</h3>
+            <p>
+              See <a href="https://developer.blackbaud.com/skyapi/apis/constituent" target="_blank">Constituent</a>
```

Comment:
We can use the updated URL here.
https://developer.blackbaud.com/skyapi/products/renxt/constituent

Reply:
Updated the URL